core/contracts/financial-templates/implementation/Liquidatable.sol

```diff
@@ -205,12 +206,15 @@ contract Liquidatable is PricelessPositionManager {
             startCollateralNetOfWithdrawal = startCollateral.sub(positionToLiquidate.withdrawalRequestAmount);
         }
 
-        FixedPoint.Unsigned memory startTokens = positionToLiquidate.tokensOutstanding;
+        // Scoping to get rid of a stack too deep error.
+        {
+            FixedPoint.Unsigned memory startTokens = positionToLiquidate.tokensOutstanding;
 
-        // Check the max price constraint to ensure that the Position's collateralization ratio hasn't increased beyond
-        // what the liquidator was willing to liquidate at.
-        // collateralPerToken >= startCollateralNetOfWithdrawal / startTokens.
-        require(collateralPerToken.mul(startTokens).isGreaterThanOrEqual(startCollateralNetOfWithdrawal));
+            // Check the max price constraint to ensure that the Position's collateralization ratio hasn't increased beyond
+            // what the liquidator was willing to liquidate at.
+            // collateralPerToken >= startCollateralNetOfWithdrawal / startTokens.
+            require(collateralPerToken.mul(startTokens).isGreaterThanOrEqual(startCollateralNetOfWithdrawal));
+        }
```

Comment:
Confused why there appear to be now extra `{`'s here

Reply:
See the comment above the first brace: this allows us to get around the `stack too deep` solidity compilation issue because it places a variable declaration in a different scope.

Reply:
I could also try removing variables, but I figured that would hurt readability more than an extra scope brace.

Reply:
ah I've never encountered this before....which variable declaration is not allowed? `startTokens`?

Reply:
You just can't use more than `n` variables in the same scope. I think `n` is 16, but I can't remember. So you either have to remove temporaries or create additional scopes/functions.

Reply:
So it doesn't matter what variable you remove, you just need to remove one.